test/multi_instance_test.go

```diff
@@ -7,97 +7,132 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-///// Tests for multiple chained instances of the protocol, no adversaries.
+// TestHonestMultiInstance_Agreement tests for multiple chained instances of the protocol with no adversaries.
+func TestHonestMultiInstance_Agreement(t *testing.T) {
+	t.Parallel()
+	const (
+		instanceCount = 4000
+		testRNGSeed   = 8965130
+		latencySeed   = testRNGSeed * 7
+	)
+	tests := []struct {
+		name        string
+		honestCount int
+		options     []sim.Option
+	}{
+		{
+			name:        "singleton sync",
+			honestCount: 1,
+			options:     syncOptions(),
+		},
+		{
+			name:        "singleton async",
+			honestCount: 1,
+			options:     asyncOptions(latencySeed),
+		},
+		{
+			name:        "pair sync",
+			honestCount: 2,
+			options:     syncOptions(),
+		},
+		{
+			name:        "pair async",
+			honestCount: 2,
+			options:     asyncOptions(latencySeed),
+		},
```

Comment:
I believe these cases are trying to cover the old TestMultiSyncAgreement. Please add (restore) node counts of 3,4,5 and 6. These small-group ones are the best for exploring different paths through the protocol steps.